torch/nn/functional.py

```diff
@@ -4790,6 +4790,18 @@ def interpolate(  # noqa: F811
         )
     if input.dim() == 5 and mode == "trilinear":
         assert align_corners is not None
+        # Two levels are necessary to prevent TorchScript from touching
+        # are_deterministic_algorithms_enabled.
+        if not torch.jit.is_scripting():
+            if torch.are_deterministic_algorithms_enabled() and (
```

Comment:
Not that it super matters, but the determinstic_algorithms call is slower than the `not input.is_cpu`. should we switch them ? 

>>> import timeit; import torch as t; x = t.tensor([1]); print(f"are_deterministic_algorithms_enabled: {timeit.timeit(lambda: t.are_deterministic_algorithms_enabled(), number=1000000):.6f}s vs is_cpu: {timeit.timeit(lambda: 
not x.is_cpu, number=1000000):.6f}s")
are_deterministic_algorithms_enabled: 0.108508s vs is_cpu: 0.062671s

I guess for cpu, it's better for the deterministic_algorithms_enabled call to go first, and for cuda the other way.. it's fine as is just thinking aloud.